### PR TITLE
Introduce FrequentStringsSketch and FrequentLongsSketch aggregation functions

### PR DESCRIPTION
**Pull request by cbalci**
Introducing two approximate aggregation function `FrequentStringsSketch` and `FrequentLongsSketch` for estimating the frequencies of items a dataset in a memory efficient way. Functions are based on the [Apache Datasketches](https://datasketches.apache.org/docs/Frequency/FrequencySketchesOverview.html) library.

**Signature:**
```
FREQUENTLONGSSKETCH(col, maxMapSize=256) -> Base64 encoded sketch object
FREQUENTSTRINGSSKETCH(col, maxMapSize=256) -> Base64 encoded sketch object
```

**Example usage:**
```
select FREQUENTSTRINGSSKETCH(AirlineID, 16) from airlineStats
```

| frequentstringssketch(AirlineID)      |
| ----------- |
| BAEKAwMAAAADAAAAAA...      |

Which can be used, for example in Java as:
```java
byte[] byteArr = Base64.getDecoder().decode(encodedSketch);
ItemsSketch<String> sketch = ItemsSketch.getInstance(Memory.wrap(byteArr), new ArrayOfStringsSerDe());

ItemsSketch.Row[] items = sketch.getFrequentItems(ErrorType.NO_FALSE_NEGATIVES);
for (int i = 0; i < items.length; i++) {
  ItemsSketch.Row item = items[i];
  System.out.printf("Airline: %s, Frequency: %d %n", item.getItem(), item.getEstimate());
}
```

**Testing:**
Basic aggregation and group by query tests are included in the PR.

**Design:**
This is a part of a larger effort to improve Datasketches support for Pinot as discussed in [this document](https://docs.google.com/document/d/1ctmKVRi67lpO6x1RYKDvDYf05EZx2Vbs2OnUudYP-bU/edit#heading=h.nctch2wugvub). Feel free to add design related comments on the document as well.

`feature` `performance`

### CODE REVIEW

**Comment by cbalci** [comment]
@chenboat @Jackie-Jiang please take a look. 
cc @mayankshriv

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+ * <p>
+ *  {@code FrequentLongsSketchAggregationFunction} provides an approximate FrequentItems aggregation function based on
+ *  <a href="https://datasketches.apache.org/docs/Frequency/FrequentItemsOverview.html">Apache DataSketches library</a>.
+ *  It is memory efficient compared to exact counting.
+ * </p>
+ * <p>
+ *   The function takes an INT or LONG column as input and returns a Base64 encoded sketch object which can be
+ *   deserialized and used to estimate the frequency of items in the dataset (how many times they appear).
+ * </p>
+ * <p><b>FREQUENT_STRINGS_SKETCH(col, maxMapSize=256)</b></p>
```
Comment (chenboat) [inline pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java:49]: Can you explain what maxMapSize means exactly in the Javadoc? Also for users what is the recommend value? Is bigger always better? Any tradeoff?

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+    LongsSketch sketch = getOrCreateSketch(aggregationResultHolder);
+
+    if (valueType == FieldSpec.DataType.BYTES) {
+      // Assuming the column contains serialized data sketch
+      LongsSketch[] deserializedSketches =
+          deserializeSketches(blockValSetMap.get(_expression).getBytesValuesSV());
+      sketch = getOrCreateSketch(aggregationResultHolder);
+
+      for (int i = 0; i < length; i++) {
```
Comment (chenboat) [inline pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java:108]: NIT: can we use foreach syntax to make it more readable?

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+  @Override
+  public void aggregate(int length, AggregationResultHolder aggregationResultHolder,
+      Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    BlockValSet valueSet = blockValSetMap.get(_expression);
+    FieldSpec.DataType valueType = valueSet.getValueType();
+
+    LongsSketch sketch = getOrCreateSketch(aggregationResultHolder);
+
+    if (valueType == FieldSpec.DataType.BYTES) {
```
Comment (chenboat) [inline pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java:102]: NIT: can we use switch-case here?

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java (hunk 1 above)
Comment (cbalci) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java:49]: Good point. Added some basic explanation to the Javadoc. I'll also follow up with a more detailed description for the function and how to use it in Pinot Docs repo.

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java (hunk 2 above)
Comment (cbalci) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java]: Done

File: pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java (hunk 3 above)
Comment (cbalci) [reply pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/FrequentLongsSketchAggregationFunction.java]: Updated

**Comment by cbalci** [comment]
Thanks for the review @chenboat. Addressed the comments, please take another look when you get a chance.

**Review by chenboat — APPROVED** [review]

**Pull request merged**